src/lib/components/AimeVideoPlayer/AimeVideoPlayer.js

```diff
@@ -1,96 +1,144 @@
-import React, { PureComponent } from "react";
+// eslint-disable-next-line no-unused-vars
+import React, { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 import ReactPlayer from "react-player";
 import {
   componentPropTypes,
   defaultComponentPropTypes,
 } from "../../utils/componentPropTypes";
+import Modal from "../modal";
+
 import styles from "./AimeVideoPlayer.module.scss";
 
-export default class AimeVideoPlayer extends PureComponent {
-  static propTypes = {
-    url: PropTypes.string.isRequired,
-    controlsVimeo: PropTypes.bool,
-    autoPlayVimeo: PropTypes.bool,
-    loopVimeo: PropTypes.bool,
-    bylineVimeo: PropTypes.bool,
-    titleVimeo: PropTypes.bool,
-    muteVimeo: PropTypes.bool,
-    backgroundVimeo: PropTypes.bool,
-    ...componentPropTypes,
-  };
+const AimeVideoPlayer = props => {
+  const {
+    theme,
+    url,
+    controlsVimeo,
+    autoPlayVimeo,
+    loopVimeo,
+    bylineVimeo,
+    titleVimeo,
+    muteVimeo,
+    backgroundVimeo,
+    withModal,
+    playsInPictureVimeo,
+  } = props;
+  const [showModal, setShowModal] = useState(false);
+  const [urlWithModal, setUrlWithModal] = useState(url);
 
-  static defaultProps = {
-    controlsVimeo: false,
-    autoPlayVimeo: false,
-    loopVimeo: false,
-    bylineVimeo: false,
-    titleVimeo: false,
-    muteVimeo: false,
-    backgroundVimeo: false,
-    ...defaultComponentPropTypes,
+  const handleModal = () => {
+    setShowModal(!showModal);
+    setUrlWithModal(url);
   };
 
-  render() {
-    const {
-      theme,
-      url,
-      controlsVimeo,
-      autoPlayVimeo,
-      loopVimeo,
-      bylineVimeo,
-      titleVimeo,
-      muteVimeo,
-      backgroundVimeo,
-    } = this.props;
+  useEffect(() => {
+    document.body.style.overflow = showModal ? "hidden" : "auto";
```

Comment:
This works, but usually you would just use `showModal` in your `render` function … or in your case, in what your functional component returns, and there add a class based on whether `showModal` is true or false.

Reply:
A little bit of vanilla JS never hurt anyone right? haha
You are 100% correct a much better approach!

Reply:
So i looked into this and actually it is the best way I can think of to manage hiding overflow-y in `theatre mode` for the body elem.
I left some comments in modal comp.
If there is a better way please show me 😃 I cant think of it.